python/ray/air/train/gbdt_trainer.py

```diff
@@ -22,7 +22,7 @@ def _convert_scaling_config_to_ray_params(
     default_ray_params: Optional[Dict[str, Any]] = None,
 ) -> "xgboost_ray.RayParams":
     default_ray_params = default_ray_params or {}
-    scaling_config_dataclass = ScalingConfigDataClass(**scaling_config)
+    scaling_config_dataclass = ScalingConfig(**scaling_config)
```

Comment:
This is no longer needed (since the input `ScalingConfig` should be the dataclass)

Reply:
Whoops you're right -- @matthewdeng should I keep support for `dict`s for backwards compatibility, or should we only support `ScalingConfig` here? (I assume the latter since `ScalingConfig` is in alpha)

Reply:
Yep your intuition is right - let's get rid of the dict altogether.